app/models/application_record.rb

```diff
@@ -7,4 +7,17 @@ class ApplicationRecord < ActiveRecord::Base
   def self.normalize(str)
     str.unicode_normalize(:nfkd).gsub(/[\u0300-\u036f]/, '').downcase
   end
+
+  def self.nilify_blank_values(*cols)
+    cols.each do |col|
+      col_changed = :"#{col}_changed?"
+      method_name = :"nilify_blank_#{col}_value"
+
+      set_callback :validation, :before, method_name, if: col_changed
+
+      define_method method_name do
+        self[col].present? || self[col] = nil
```

Comment:
I'm not a fan of using boolean logic for control flow like this. `self[col] = nil if self[col].blank?` is a lot clearer, IMO.

Reply:
agreed. I had a very similar method in a different project, but had to read this part twice to remember how it worked 😅